pghoard/basebackup.py

```diff
@@ -110,28 +115,36 @@ def compress_directly_to_a_file(self, proc, basebackup_path):
                 stderr=proc.stderr,
                 output_obj=output_obj,
                 compression_algorithm=compression_algorithm,
-                rsa_public_key=rsa_public_key)
+                rsa_public_key=rsa_public_key,
+                stderr_deque=stderr_deque)
             os.link(output_obj.name, basebackup_path)
 
         metadata = {
             "compression-algorithm": compression_algorithm,
             "encryption-key-id": encryption_key_id,
         }
-        return original_input_size, compressed_file_size, metadata
+        return original_input_size, compressed_file_size, metadata, stderr_deque
+
+    def read_stdout_stderr(self, proc, stderr_deque):
+        rlist, _, _ = select.select([proc.stdout, proc.stderr], [], [], 1.0)
+        for fd in rlist:
+            content = fd.read()
+            if content:
+                stderr_deque.append(content)
```

Comment:
considering the limit is called `LINES_OF_OUTPUT_TO_STORE`  we probablywant stderr_deque.extend(content.splitlines())` instead?